docs/tutorials/porch-development-environment/bin/setup.sh

```diff
@@ -14,93 +14,110 @@
+h1 Install Gitea
+mkdir "$TMP_DIR/kpt_packages"
+cd "$TMP_DIR/kpt_packages"
 kpt pkg get https://github.com/nephio-project/catalog/tree/main/distros/sandbox/gitea
-$SED -i 's/ metallb.universe.tf/ #metallb.universe.tf/' gitea/service-gitea.yaml
+kpt fn eval gitea \
+  --image gcr.io/kpt-fn/set-annotations:v0.1.4 \
+  --match-kind Service \
+  --match-name gitea \
+  --match-namespace gitea \
+  -- "metallb.universe.tf/loadBalancerIPs=${gitea_ip}"
+curl -o gitea/cluster-config.yaml https://raw.githubusercontent.com/nephio-project/porch/main/docs/tutorials/starting-with-porch/kind_management_cluster.yaml
+echo "metadata: { name: "porch-test" }" >> gitea/cluster-config.yaml
+kpt fn eval gitea \
+  --image gcr.io/kpt-fn/set-annotations:v0.1.4 \
+  --match-kind Cluster \
+  --match-api-version kind.x-k8s.io/v1alpha4 \
+  -- "config.kubernetes.io/local-config=true"
+kpt fn eval gitea \
+  --image gcr.io/kpt-fn/apply-replacements:v0.1.1 \
+  --fn-config "${self_dir}/replace-gitea-service-ports.yaml"
+
 kpt fn render gitea
 kpt live init gitea
 kpt live apply gitea
 
-popd || exit
-
-# Create management and edge1 repos in gitea
-curl -k -H "content-type: application/json" "http://nephio:secret@172.18.255.200:3000/api/v1/user/repos" --data '{"name":"management"}'
-curl -k -H "content-type: application/json" "http://nephio:secret@172.18.255.200:3000/api/v1/user/repos" --data '{"name":"edge1"}'
-
-mkdir repos
-pushd repos || exit
-
-# Initialize management and edge1 repos in Gitea
-git clone http://172.18.255.200:3000/nephio/management
-pushd management || exit
-
-touch README.md
-git init
-git checkout -b main
-git config user.name nephio
-git add README.md
-
-git commit -m "first commit"
-git remote remove origin
-git remote add origin http://nephio:secret@172.18.255.200:3000/nephio/management.git
-git remote -v
-git push -u origin main
-popd || exit
-
-git clone http://172.18.255.200:3000/nephio/edge1
-pushd edge1 || exit
-
-touch README.md
-git init
-git checkout -b main
-git config user.name nephio
-git add README.md
-
-git commit -m "first commit"
-git remote remove origin
-git remote add origin http://nephio:secret@172.18.255.200:3000/nephio/edge1.git
-git remote -v
-git push -u origin main
-popd || exit
+############################################
+h1 Create git repos in gitea
+curl -k -H "content-type: application/json" "http://nephio:secret@${gitea_ip}:3000/api/v1/user/repos" --data "{\"name\":\"$git_repo_name\"}"
+
+mkdir "$TMP_DIR/repos"
+cd "$TMP_DIR/repos"
+git clone "http://nephio:secret@${gitea_ip}:3000/nephio/$git_repo_name"
+cd "$git_repo_name"
+
+if ! git rev-parse -q --verify refs/remotes/origin/main >/dev/null; then
+  git switch -c  main
+  touch README.md
+  git add README.md
+  git config user.name nephio
+  git commit -m "first commit"
+  git push -u origin main
+else
+  echo "main branch already exists in git repo."
+fi
 
-popd || exit
+############################################
+h1 Generate certs and keys
```

Comment:
Would you not prefer to patch this into #43

Reply:
sure. I just wanted to keep new "features" (steps) separate from the refactor part. but I don't mind at all to merge the two.

Reply:
Up to you. I did run this PR locally and the makekeys executes but I don't see the certs anywhere.

Reply:
thanks, that is actually a bug. I fixed it below